### ISSUE TITLE
Training two NN simultaneously.

### ISSUE THREAD

**Issue by MitchVonk** [keras-team/keras#10121]
Is it possible to train two different NN simultaneously, where the output of the first NN is partially the input for the second NN?

**Comment by MitchVonk** [comment]
Or have more input values in a deeper layer?

**Comment by nittanycolonial** [comment]
Are you able to post a use case? That might help people guide you a little more

**Comment by MitchVonk** [comment]
Lets say I have two models, model1 and model2:

model1 = Sequential([
        Dense(20, input_shape=(10,)),
        Dense(20, activation='relu'),
        Dense(2, activation='softmax')
            ])
model2 = Sequential([
        Lambda(norm_input, input_shape=(1,28,28), output_shape=(1, 28,28)),
        Conv2D(32,(3,3), activation='relu'),
        Flatten(),
        Dense(2, activation='softmax')
        ])

Now I want the output the conv model2 be one of the 10 input values of model1, and then train both models at the same time. (Thus I have two types of data sets: 9x1 (10x1 with output of model 10) and 28x28)

**Comment by maderafunk** [comment]
I have not tested it, but its possible with the functional api:
https://keras.io/getting-started/functional-api-guide/


```
from keras.layers import Input, Dense
from keras.models import Model
from keras.layers import concatenate

input_1 = Input(shape=(1,28,28))
input_2 = Input(shape=(9,))

x_1 = Conv2D(32,(3,3), activation='relu')(input_1 )
x_1 = Flatten()(x_1 )
predictions_1 = Dense(1, activation='sigmoid')(x_1 )

merged_vector = concatenate([predictions_1 , input_2 ])
x_2 = Dense(20)(merged_vector)
x_2 = Dense(20, activation='relu')(x_2)
predictions_2 =Dense(2, activation='softmax')(x_2)

model = Model(inputs=[input_1,input_2] , outputs=[predictions_1 ,predictions_2])

# or alternatively with only one loss:
# model = Model(inputs=[input_1,input_2] , outputs=[predictions_1])

model.compile(optimizer='rmsprop',
              loss=['binary_crossentropy','categorical_crossentropy'],
              metrics=['accuracy'])
model.fit([data_1,data_2], [labels_1, labels_2]) 
```
